File: scala/src/main/scala/nl/quintor/akkastreams/StreamingNumbers.scala
package nl.quintor.akkastreams

import akka.actor.ActorSystem
import akka.stream.scaladsl._
import akka.{Done, NotUsed}

import scala.concurrent.Future
import scala.concurrent.duration.DurationInt

object StreamingNumbers extends App {
  implicit val system: ActorSystem = ActorSystem("Workshop")

  val numberSource: Source[Int, NotUsed] = Source(1 to 20)
  val higherNumbers: Source[Int, NotUsed] = Source(20 to 40)

  //Print all numbers  to console

  //Print all even numbers

  //Print the sum of all even numbers and the sum of all uneven numbers

  //Print pairs of numbers from both sources

  //Print one number per second

}

File: java/src/main/java/nl/quintor/akkastreams/StreamingNumbers.java
package nl.quintor.akkastreams;

import akka.actor.ActorSystem;
import akka.stream.javadsl.Source;
import java.time.Duration;

public class StreamingNumbers {

  public static void main(String[] args) {
    final ActorSystem system = ActorSystem.create("QuickStart");

    var numberSource = Source.range(1, 20);
    var higherNumbers = Source.range(20, 40);

    //Print all numbers  to console

    //Print all even numbers

    //Print the sum of all even numbers and the sum of all uneven numbers

    //Print pairs of numbers from both sources

    //Print one number per second

  }
}